src/Security/Voter/HandleCommandVoter.php

```diff
@@ -64,13 +64,15 @@ protected function supports($attribute, $subject)
     protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
     {
         $allowedRoles = $this->getAllowedRoles(get_class($subject));
+        $actualRoles = $this->roleHierarchy->getReachableRoles($token->getRoles());
 
-        if (count($allowedRoles) > 0) {
-            return $this->decisionManager->decide($token, $allowedRoles);
-        } else {
-            // default conclusion is access denied
-            return false;
+        foreach ($actualRoles as $role) {
+            if (in_array($role->getRole(), $allowedRoles)) {
+                return true;
+            }
         }
```

Comment:
Looking at http://symfony.com/doc/current/components/security/authorization.html#rolehierarchyvoter, that's not the way symfony performs role checks as far as I can see. So, you would end up with a system that performs role checks in different manners, or am I missing something here?